astropy/time/core.py

```diff
@@ -1787,6 +1787,85 @@ def light_travel_time(self, skycoord, kind='barycentric', location=None, ephemer
+        Adds longitude including the TIO locator rigorously corrected
+        for polar motion.
+
+        Parameters
+        ---------------
+        theta : float or float array
+            Earth rotation angle or Greenwich sidereal time in radians.
+        longitude : `~astropy.coordinates.Longitude`
+            The longitude on the Earth at which to compute the result.
+
+        Returns
+        -------
+        local_angle : float or float array
+            Either local Earth rotation angle or local sidereal time depending
+            on theta.
+        """
+
+        from astropy.coordinates.builtin_frames.utils import get_polar_motion
+        from astropy.coordinates.matrix_utilities import rotation_matrix
+
+        jd1_tt = getattr(self, 'tt')._time.jd1
+        jd2_tt = getattr(self, 'tt')._time.jd2_filled
+        sp = erfa.sp00(jd1_tt, jd2_tt)
+        xp, yp = get_polar_motion(self._time)
+        lon = longitude.lon.to_value(u.radian)
+        # Form the rotation matrix, CIRS to apparent [HA,Dec].
+        r = (rotation_matrix(lon, 'z', unit=u.radian)
+             @ rotation_matrix(-yp, 'x', unit=u.radian)
+             @ rotation_matrix(-xp, 'y', unit=u.radian)
+             @ rotation_matrix(theta+sp, 'z', unit=u.radian))
+
+        # Solve for local angle.
+        a = r[..., 0, 0]
+        b = r[..., 0, 1]
+        local_angle = np.arctan2(b, a)
+        return local_angle
+
+    def local_earth_rotation_angle(self, longitude=None):
+        """ Calculate local Earth rotation angle.
+
+        Parameters
+        ---------------
+        longitude : `~astropy.units.Quantity`, `str`, or None; optional
+            The longitude on the Earth at which to compute the Earth rotation
+            angle. Can be given as a `~astropy.units.Quantity` with angular
+            units (or an `~astropy.coordinates.Angle` or
+            `~astropy.coordinates.Longitude`), or as a name of an
+            observatory (currently, only ``'greenwich'`` is supported,
+            equivalent to 0 deg).  If `None` (default), the ``lon`` attribute
+            of the Time object is used.
+
+        Returns
+        -------
+        eral : `~astropy.coordinates.Longitude`
+            Local Earth rotation angle as a quantity with units of hourangle
+        """
+
+        from astropy.coordinates import Longitude
+
+        if longitude is None:
+            if self.location is None:
+                raise ValueError('No longitude is given but the location for '
+                                 'the Time object is not set.')
+            longitude = self.location.lon
+        elif longitude == 'greenwich':
+            longitude = Longitude(0., u.degree,
+                                  wrap_angle=180. * u.degree)
+        else:
+            # sanity check on input
+            longitude = Longitude(longitude, u.degree,
+                                  wrap_angle=180. * u.degree)
+
+        jd1_ut1 = getattr(self, 'ut1')._time.jd1
```

Comment:
Here, using `get_jd12` is probably not work it, but it can be a bit simpler:
```
jd1_ut1 = self.ut1._time.jd1
jd2_ut2 = self.ut1._time.jd2_filled
```

Reply:
I don't see why not. In fact, it's probably more important here than in the case of `tt`, see above.